dex/msgjson/types.go

```diff
@@ -828,7 +829,7 @@ type OrderBook struct {
 type MatchProofNote struct {
 	MarketID  string  `json:"marketid"`
 	Epoch     uint64  `json:"epoch"`
-	Preimages []Bytes `json:"preimages"`
+	Preimages []order.Preimage `json:"preimages"`
```

Comment:
Is it okay to change this type ? Changing this type seems only natural to me, to avoid unnecessary type-conversions when finding relevant tracked orders in `validateMatchCsum` func.

Reply:
Depends on how it marshals/unmarshals to JSON.  I don't believe it implements either json.Marshaller or Unmarshaller, so I suspect it's going to be an array of integers as a result instead of hex.

We lack a TestMatchProof in msg_test, but you could make one to see if it marshals and unmarshals as expected.

```go
func TestMatchProof(t *testing.T) {
	pimgB := randomBytes(32)
	var pimg order.Preimage
	copy(pimg[:], pimgB)
	match := &MatchProofNote{
		Preimages: []order.Preimage{pimg},
	}

	matchB, err := json.Marshal(match)
	if err != nil {
		t.Fatalf("marshal error: %v", err)
	}

	t.Logf("%v", string(matchB))

    // check things here

	var matchBack Match
	err = json.Unmarshal(matchB, &matchBack)
	if err != nil {
		t.Fatalf("unmarshal error: %v", err)
	}

    // check more things here
}
```

> === RUN   TestMatchProof
>     /home/jon/github/decred/dcrdex/dex/msgjson/msg_test.go:83: {"marketid":"","epoch":0,"preimages":[[82,253,252,7,33,130,101,79,22,63,95,15,154,98,29,114,149,102,199,77,16,3,124,77,123,187,4,7,209,226,198,73]],"misses":null,"csum":"","seed":""}
> 

With `[]Bytes`:

> === RUN   TestMatchProof
>     /home/jon/github/decred/dcrdex/dex/msgjson/msg_test.go:80: {"marketid":"","epoch":0,"preimages":["52fdfc072182654f163f5f0f9a621d729566c74d10037c4d7bbb0407d1e2c649"],"misses":null,"csum":"","seed":""}

Reply:
Rolled back type change.